language/java/language.go

```diff
@@ -88,6 +88,16 @@ func (l *Language) TestFramework() (testFramework string) {
 	return "JUnit 5"
 }
 
+// DefaultFileExtension returns the default file extension.
+func (l *Language) DefaultFileExtension() string {
+	return ".java"
+}
+
+// DefaultTestFileSuffix returns the default test file suffix.
+func (l *Language) DefaultTestFileSuffix() string {
+	return "Test.java"
+}
+
```

Comment:
nit: for java we sometimes also have `Tests.java` so since this is just for filtering we could also say `IsTestFilePath` and use a regex that allows for both. But since we decide how the examples look, it's fine!